openmmtools/multistate/multistatesampler.py

```diff
@@ -1758,12 +1758,10 @@ def _update_timing(self, iteration_time, partial_total_time, run_initial_iterati
         # Estimate performance
         # TODO: use units for timing information to easily convert between seconds and days
         # there are some mcmc_moves that have no timestep attribute, catch exception
-        try:
-            current_simulated_time = self.mcmc_moves[0].timestep * self._iteration * self.mcmc_moves[0].n_steps
-        except AttributeError:
-            current_simulated_time = 0.0 * unit.nanosecond  # Hardcoding to 0 ns
-        performance = current_simulated_time.in_units_of(unit.nanosecond) / (partial_total_time / 86400)
-        self._timing_data["ns_per_day"] = performance.value_in_unit(unit.nanosecond)
+        moves_iterator = self._flatten_moves_iterator()
+        current_simulated_nanoseconds = sum([move.timestep.value_in_unit(unit.nanosecond) * move.n_steps for
+                                             move in moves_iterator if hasattr(move, "timestep") and hasattr(move, "n_steps")])
+        self._timing_data["ns_per_day"] = current_simulated_nanoseconds / (partial_total_time / 86400)
```

Comment:
Might want to add something like `86400 (seconds in a day)` or something -- not everyone will know what that number means. Even better would be to use `unit` to do the conversion since we are using the line above it.